roles/content_views/tasks/_create_content_view.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: parse names to proper repo names
+  set_fact:
+    content_temp_names: "{{
+        (content.repos | map(attribute='name') | map('regex_replace', '[()]')) |
+          zip(
+            (content.repos | map(attribute='basearch', default='MISSING')),
+            (content.repos | map(attribute='releasever', default='MISSING'))
+          ) |
+          map('join', ' ') | map('regex_replace', ' MISSING') | list
+      }}"
+    content_temp_repositories: []
+  no_log: "{{ foreman_configuration_content_views_secure_logging }}"
+
+- name: create list of repos
+  set_fact:
+    content_temp_repositories: "{{ content_temp_repositories + [{ 'name': content_temp_names[item.0], 'product': item.1.product}] }}"
+  with_indexed_items: "{{ content.repos }}"
+  no_log: "{{ foreman_configuration_content_views_secure_logging }}"
+
+- name: create content view
+  theforeman.foreman.content_view:
+    username: "{{ username | default(omit) }}"
```

Comment:
Here and throughout, I think you can now update these to the scoped variables, e.g. `foreman_username`